src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -1,90 +1,58 @@
 package core.basesyntax;
 
-import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 public class JavaStreamApi {
 
-    /**
-     * <p>1. Дано: List of Integer numbers.
-     * Вернуть сумму нечетных числел или 0, если таких несуществует</p>
-     **/
     public Integer oddSum(List<Integer> numbers) {
-        return null;
+        return numbers.stream().filter((x) -> x % 2 != 0).reduce(Integer::sum).orElse(0);
     }
 
-    /**
-     * <p>2. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть количество вхождений объекта `element`</p>
-     **/
     public Long elementCount(List<String> elements, String element) {
-        return null;
+        return elements.stream().filter(element::equals).count();
     }
 
-    /**
-     * <p>3. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть Optional первого элемента коллекции</p>
-     **/
     public Optional<String> firstElement(List<String> elements) {
-        return null;
+        return elements.stream().findFirst();
     }
 
-    /**
-     * <p>4. Дана коллекция строк List of String elements
-     * (приметр: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException</p>
-     **/
     public String findElement(List<String> elements, String element) {
-        return null;
+        return elements.stream().filter(element::equals).findFirst().get();
```

Comment:
```suggestion
        return elements.stream()
        .filter(element::equals)
        .findFirst()
        .orElseThrow();
```
result will be the same, but this way it will give you more informations about what should happen

Reply:
done